File: 0554/solution.java
import java.util.Map;
import java.util.HashMap;
import java.util.List;

class Solution {
    public int leastBricks(List<List<Integer>> wall) {
        Map<Integer, Integer> cuts = new HashMap<>();
        for(List<Integer> row: wall) {
            int total = 0;
            for(int i = 0; i < row.size() - 1; i++) {
                total += row.get(i);
                cuts.put(total, cuts.getOrDefault(total, 0) + 1);
            }
        }
        int res = 0;
        for(int value: cuts.values()) {
            res = Math.max(value, res);
        }
        return wall.size() - res;
    }
}

File: 0554/solution.py
from collections import defaultdict


class Solution:
    def leastBricks(self, wall: list[list[int]]) -> int:
        cuts = defaultdict(int)
        for row in wall:
            total = 0
            for brick in row[:-1]:
                total += brick
                cuts[total] += 1
        if not cuts:
            return len(wall)
        return len(wall) - max(cuts.values())
